docs/features/nullable-metadata.md

```diff
@@ -0,0 +1,125 @@
+4. All parts are not annotated and the [context](#NullableContextAttribute) is oblivious: `NullableAttribute(1)` (rather than `new byte[] { 1 }`)
+5. All parts are annotated: `NullableAttribute(2)` (rather than `new byte[] { 2 }`)
+
+The first case, where all parts are oblivious in an oblivious context, matches the representation in legacy assemblies where there are no `NullaleAttribute` annotations.
+
+`NullableAttribute(1)` should be placed on a type parameter definition that has a `class!` constraint.
+`NullableAttribute(2)` should be placed on a type parameter definition that has a `class?` constraint.
+`NullableAttribute(2)` should be placed on a type parameter definition that has no type constraints, `class`, `struct` and `unmanaged` constraints and
+is declared in a context where nullable type annotations are allowed, that is eqivalent to having an `object?` constraint.
+Other forms of `NullableAttribute` are not emitted on type parameter definitions and are not specially recognized on them.
+
+The type declaration for `NullableAttribute` is synthesized by the compiler if not already included in the compilation.
+
+### NullableContextAttribute
+`NullableContextAttribute` can be used to indicate the nullability of type references that have no `NullableAttribute` annotations.
+
+```C#
+namespace System.Runtime.CompilerServices
+{
+    [System.AttributeUsage(
+        AttributeTargets.Assembly |
+        AttributeTargets.Class |
+        AttributeTargets.Delegate |
+        AttributeTargets.Interface |
+        AttributeTargets.Struct |
+        AttributeTargets.Event |
+        AttributeTargets.Field |
+        AttributeTargets.Method |
+        AttributeTargets.Property,
+        AllowMultiple = false,
+        Inherited = false)]
+    public sealed class NullableContextAttribute : Attribute
+    {
+        public readonly bool Enabled;
+        public NullableContextAttribute(bool enabled)
+        {
+            Enabled = enabled;
+        }
+    }
+}
+```
+
+The `NullableContextAttribute` is optional - nullable annotations can be represented in metadata with full fidelity using `NullableAttribute` only.
+
+`NullableContextAttribute` can be added at different levels of the metadata hierarchy to specify how reference types
+without a `NullableAttribute` are interpreted.
+The attribute constructor takes a `bool enabled` argument:
+if `true`, reference types are interpreted as not nullable;
+if `false`, reference types are interpreted as oblivious.
+If there is no `NullableContextAttribute` on the containing member or containing types or assembly,
+unannotated reference types are interpreted as oblivious.
+
+The attribute is not inherited.
+
+The C#8 compiler will recognize `NullableContextAttribute` instances on the assembly, types (top-level and nested types),
+and members.
+`NullableContextAttribute` will be ignored on other targets.
```

Comment:
Is including other targets as valid in attribute definition safe? I think it is better to disallow the attributes on targets that we do not plan to recognize.